src/compile/scale/domain.ts

```diff
@@ -166,85 +160,89 @@ function parseSingleChannelDomain(scaleType: ScaleType, domain: Domain, model: U
     } else {
       if (isDateTime(domain[0])) {
         return (domain as DateTime[]).map((dt) => {
-          return {signal: dateTimeExpr(dt, true)};
+          return {signal: `{data: ${dateTimeExpr(dt, true)}}`};
         });
       }
-      return domain;
+      return [domain];
     }
   }
 
   const stack = model.stack;
   if (stack && channel === stack.fieldChannel) {
     if(stack.offset === 'normalize') {
-      return [0, 1];
+      return [[0, 1]];
     }
-    return {
-      data: model.requestDataName(MAIN),
-      fields: [
-        model.field(channel, {suffix: 'start'}),
-        model.field(channel, {suffix: 'end'})
-      ]
-    };
+
+    const data = model.requestDataName(MAIN);
+    return [{
```

Comment:
I disagree with the second part. I want the domain to be an array as until assemble because that makes merging very easy. During parse, we only concat and during assembly, we don't have to deal with unioned domains. 

I also don't want to rename the method because the method is parsing the domain (but happens to return an array of domain objects, which represent the domain.